addons/website/models/ir_http.py

```diff
@@ -443,6 +443,9 @@ def get_frontend_session_info(self):
             session_info.update({
                 'website_id': request.website.id,
                 'website_company_id': request.website._get_cached('company_id'),
+                # We need the user_context if we want the configurator 
+                # to be in the user's language.
+                'user_context': request.session.get_context(),
```

Comment:
About this commit, the code of the configurator seems to be so unusual, a bit of technical explanation might be useless here.. Or I/we have to redo all the technical investigation you probably did to figure stuff and ensure the fix is right.

In this case, correct me if I am wrong:
- We have a website=True controller, which then need to add a lang to its call to `render()` to avoid using the frontend lang (frontend lang is obviously expected as it is a website=True controller, so a frontend lang) 
- That controller returns an empty page (empty body) which is then set in JS (not only the dynamic content, but the static sentences too like "Ready to build your website?" :thinking: )

So you need to do this kind of black magic (which is a generic fix used only for this weird controller, right?) to alter the Odoo normal behavior of the frontend, to translate the content it gets through RPC in another language than the one of the frontend.
It just seems like we are on a frontend controller but actually just want it to act as a backend controller :shrug: 

I wonder why this `/website/configurator` controller is set as frontend and actually inherit the frontend_layout instead of just being a regular backend controller.
```xml
<template id="website_configurator" name="Website Configurator" inherit_id="web.frontend_layout" primary="True">
    <xpath expr="//head/t[@t-call-assets][last()]" position="after">
        <t t-call-assets="website.website_configurator_assets_js" lazy_load="True"/>
    </xpath>
    <xpath expr="//t[@t-call-assets='web.assets_frontend']" position="replace">
        <t t-call-assets="website.website_configurator_assets_scss" t-js="false"/>
    </xpath>
    <xpath expr="//nav" position="replace"/>
    <xpath expr="//div[@id='wrapwrap']" position="replace"/>
</template>
```
There should be an obvious reason, but I really never dig into the configurator code.

Not saying your fix is wrong, just trying to figure stuff here.

Anyway, to sum up your fix, you need it for `localizationService`, which depends of `userService`, to get the wanted lang, from the context?

```js
get lang() {
    return context.lang;
},
```

Reply:
The summarisation is correct. I needed the lang in the context for the localizationService to work properly. 

I'm not sure I understand what Website=true does. I tried looking in the code but I couldn't find a direct answer. Do you happen to know where I could find infos about that? I could then check the configurator code and have an answer as to why it's frontend.

In any case, it might be interesting to move it to the backend in master if task-2687506 gets merged (move edit to backend), because with the current code, I think the user would : create a new website / install from backend -> move to front for the configurator -> move to backend to edit.